cpp/src/skyhook/protocol/skyhook_protocol.cc

```diff
@@ -0,0 +1,139 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+#include "skyhook/protocol/skyhook_protocol.h"
+
+#include <flatbuffers/flatbuffers.h>
+
+#include "arrow/io/api.h"
+#include "arrow/ipc/api.h"
+#include "arrow/result.h"
+
+#include "ScanRequest_generated.h"
+
+namespace skyhook {
+
+namespace flatbuf = org::apache::arrow::flatbuf;
+
+arrow::Status SerializeScanRequest(ScanRequest req, ceph::bufferlist& bl) {
+  auto filter_expression = arrow::compute::Serialize(req.filter_expression).ValueOrDie();
+  auto partition_expression =
+      arrow::compute::Serialize(req.partition_expression).ValueOrDie();
+  auto projection_schema =
+      arrow::ipc::SerializeSchema(*req.projection_schema).ValueOrDie();
+  auto dataset_schema = arrow::ipc::SerializeSchema(*req.dataset_schema).ValueOrDie();
```

Comment:
Replace `ValueOrDie` with `ARROW_ASSIGN_OR_RAISE`

Reply:
Looks like `ARROW_ASSIGN_OR_RAISE` is not available in `arrow` namespace.

Reply:
Can you please suggest some other solution ?

Reply:
You shouldn't have to import anything.  It's actually used (I think from some of my suggestions) in this file already.  I've submitted a PR that should address the remaining cases.  Please take a look and see how it handles errors more gracefully.

Reply:
Thanks a lot @westonpace . I have merged the changes.